src/features/Send/useCases/ListSelectedTokens/EditAmount/EditAmountScreen.tsx

```diff
@@ -0,0 +1,210 @@
+import {useNavigation} from '@react-navigation/native'
+import * as React from 'react'
+import {
+  KeyboardAvoidingView,
+  Platform,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  ViewProps,
+} from 'react-native'
+
+import {Button, Spacer, TextInput} from '../../../../../components'
+import {AssetItem} from '../../../../../components/AssetItem'
+import {TxHistoryRouteNavigation} from '../../../../../navigation'
+import {useSelectedWallet} from '../../../../../SelectedWallet'
+import {COLORS} from '../../../../../theme'
+import {PairedBalance} from '../../../../../TxHistory/PairedBalance'
+import {useTokenInfo} from '../../../../../yoroi-wallets/hooks'
+import {Logger} from '../../../../../yoroi-wallets/logging'
+import {Quantity} from '../../../../../yoroi-wallets/types'
+import {asQuantity, Quantities} from '../../../../../yoroi-wallets/utils'
+import {editedFormatter, pastedFormatter} from '../../../../../yoroi-wallets/utils/amountUtils'
+import {useTokenQuantities} from '../../../common/hooks'
+import {useSend} from '../../../common/SendContext'
+import {useStrings} from '../../../common/strings'
+import {NoBalance} from './ShowError/NoBalance'
+import {UnableToSpend} from './ShowError/UnableToSpend'
+
+export const EditAmountScreen = () => {
+  const strings = useStrings()
+  const navigation = useNavigation<TxHistoryRouteNavigation>()
+  const {selectedTokenId, amountChanged} = useSend()
+  const {available, spendable, initialQuantity} = useTokenQuantities(selectedTokenId)
+
+  const wallet = useSelectedWallet()
+  const tokenInfo = useTokenInfo({wallet, tokenId: selectedTokenId})
+  const isPrimary = tokenInfo.id === wallet.primaryTokenInfo.id
+
+  const [quantity, setQuantity] = React.useState<Quantity>(initialQuantity)
+  const [inputQuantity, setInputQuantity] = React.useState<Quantity>(
+    Quantities.denominated(initialQuantity, tokenInfo.decimals),
+  )
+
+  const hasBalance = !Quantities.isGreaterThan(quantity, available)
+  const isUnableToSpend = isPrimary && Quantities.isGreaterThan(quantity, spendable)
+  const isZero = Quantities.isZero(quantity)
+
+  const onChangeQuantity = (text: string) => {
+    try {
+      const quantity = asQuantity(text)
+      setInputQuantity(quantity)
+      setQuantity(Quantities.fixed(quantity, tokenInfo.decimals))
```

Comment:
what is Quantities.fixed?

Reply:
`1.1` (6 d) -> `1100000`, -> toIntString?

Reply:
renamed to integer